api/src/main/java/jakarta/servlet/http/Cookie.java

```diff
@@ -441,9 +441,19 @@ public boolean isHttpOnly() {
      * except for <code>version</code>. E.g. when <code>cookie.setAttribute("domain", domain)</code> is invoked, then
      * <code>cookie.getDomain()</code> should return exactly that value, and vice versa.
      *
+     * <p>
+     * Attributes with a value of the empty string will appear as {@code "attribute-name="} in any {@code Set-Cookie} header
+     * generated from the {@code Cookie} instance unless special handling is required as defined in the following paragraph.
+     *
+     * <p>
+     * Attributes that are defined as not having a value, such as {@code HttpOnly}, must set with a value such that {@code
+     * Boolean.parseBoolean(attribute-value)} evaluates to {@code true} for the attribute to be included in any
+     * {@code Set-Cookie} header generated from the {@code Cookie} instance, otherwise the attribute is not included.
```

Comment:
This whole block can be dropped if the verbiage above is done instead.

This makes the results of `setAttribute("name", "")` produce parsing rules that comply today with things like `Partitioned`, and any future attributes without values too.

Perhaps this can even provide a code block example.

``` java
<code>
setAttribute("name", "value"); // becomes "name=value"
setAttribute("name", ""); // becomes "name"
setAttribute("name", null); // removes "name" from the cookie
setAttribute("HttpOnly", ""); // becomes "HttpOnly"
setAttribute("Partitioned", ""); // becomes "Partitioned"
setAttribute("SameSite", "Strict"); // becomes "SameSite=Strict"
</code>
```

Pointing out what happens?